internal/ast/types.go

```diff
@@ -0,0 +1,16 @@
+package ast
+
+type Meta struct {
+	Position int `yaml:"pos"`
+}
+
+type TernaryOperator [3]Node
+type VariadicOperator []Node
+
+type Node struct {
+	Meta     `yaml:",inline"`
+	EqFilter *TernaryOperator  `yaml:"eqfilter,omitempty"`
+	Path     *VariadicOperator `yaml:"path,omitempty"`
+	String   *string           `yaml:"string,omitempty"`
```

Comment:
`Interger` and `Boolean` are missing. I know they are not required to implement the simple routes filtering case.
But we are configuring the API and we know they will be needed. I will add them in this PR. But up to you, can be done in a follow-up PR.